cms/bundles/forms.py

```diff
@@ -61,6 +73,68 @@ def _has_datasets(self) -> bool:
 
         return has_datasets
 
+    def _get_dataset_title_from_form_data(self, dataset_id: str) -> str:
+        """Get dataset title from form data based on dataset ID."""
+        for form in self.formsets["bundled_datasets"].forms:
+            if (
+                form.is_valid()
+                and not form.cleaned_data.get("DELETE")
+                and (dataset := form.cleaned_data.get("dataset"))
+                and dataset.namespace == dataset_id
+            ):
+                return str(dataset.title)
+        return dataset_id
+
+    def _check_bundle_contents_approval_status(self) -> list[str]:
+        """Check bundle contents and return list of non-approved datasets."""
+        client = BundleAPIClient()
+        datasets_not_approved: list[str] = []
+
+        # Ensure bundle_api_id is not None
+        if not self.instance.bundle_api_id:
+            return datasets_not_approved
+
+        try:
+            response = client.get_bundle_contents(self.instance.bundle_api_id)
+
+            # Check each content item in the bundle
+            for item in response.get("contents", []):
+                item_state = item.get("state", "unknown")
+
+                if item_state != "APPROVED":
```

Comment:
How about
```suggestion
                if item_state != BundleStatus.APPROVED:
```

? since the states map directly to BundleStatus. Alternatively, we have a `BundleState` enum in the client

Reply:
Changed in [fb5974f](https://github.com/ONSdigital/dis-wagtail/pull/305/commits/fb5974ff7aed0e47ac533e20f2cd6087d06261a4)